lib/modules/download/generators/shapefile.rb

```diff
@@ -104,6 +104,11 @@ def merge_files
     range = (0..@number_of_pieces-1)
     files_paths = range.map { |i| zip_path(i) }.join(' ')
     system("zip -j #{zip_path} #{files_paths}") and system("zip -ru #{zip_path} *", chdir: ATTACHMENTS_PATH)
+    system("zip -ru #{zip_path} #{File.basename(sources_path)}", chdir: File.dirname(sources_path))
     range.each { |i| FileUtils.rm_rf(zip_path(i)) }
   end
+
+  def sources_path
+    File.join(File.dirname(zip_path), "WDPA_sources.csv")
+  end
```

Comment:
It seems like this method already exists in the base class this one inherits from?
Perhaps you could just assign `zip_path` to an instance variable with the same name so that it can be used by the method in the base class? @lucacug

Reply:
@lucacug Also just realised it's worth considering using the already existing method in the base class named `export_sources`, which creates the file first if it doesn't exist at the moment of generating the download